### ISSUE TITLE
 Invalid parameter provided for 'modrinth' command: No files are available for the project fRQREgAc for loader paper and Minecraft version 1.20.6

### ISSUE THREAD

**Issue by x0tester0x** [itzg/mc-image-helper#428]
This is the configuration in the docker compose file:

```
- TYPE=PAPER
- VERSION=LATEST
 - MODRINTH_PROJECTS=mcdash,maintenance,webstats,dynmap,huskhomes,sleeper
```

What does not work here, because after this error the minecraft server stops?

**Comment by itzg** [comment]
It means one of those plugins is not available for Paper 1.20.6. Unfortunately the log is showing the Modrinth project ID rather than the slug you provided so I'll have to lookup an easy way for you to convert that identifier.

**Comment by x0tester0x** [comment]
**xyz** is a placeholder...
OK, but why then stop the server?

**Comment by itzg** [comment]
It stops the startup because you misconfigured it. Otherwise if it let the misconfiguration proceed then it would get lost in the logs and you'd wonder why the plugin was missing.

**Comment by x0tester0x** [comment]
OK, but when there exists already a plugin.jar in the plugins folder with version 1.20.4 and then the paper.jar is updated from 1.20.4 to 1.20.6 what happens then?

**Comment by itzg** [comment]
It will attempt to upgrade the plugin by downloading the new and removing the old.

**Comment by x0tester0x** [comment]
OK, but in the case, when no new verison of the plugin is found, the server does not stop or?

**Comment by itzg** [comment]
Correct. It's trying to find a version of that plugin compatible with Paper 1.20.6, not finding one, and stopping the startup process so it's obvious there is something you need to fix.

**Comment by x0tester0x** [comment]
OK in this case VERSION=LATEST will always stop the server so I have to set a fixed version...

Can I set in the configuration that it should use not the latest Minecraft Client release but the latest stable Paper Server release?
```
- TYPE=PAPER
- VERSION=LATEST
```

**Comment by itzg** [comment]
It's best to always set a specific VERSION, such as 1.20.4 in this case. 

Narrowing the Paper release channel is a planned feature https://github.com/itzg/docker-minecraft-server/issues/2811 ; however, even with that it will still leave room for surprise upgrades.

**Comment by itzg** [comment]
@Jannled not about a "version check". The automation is using Modrinth's API to lookup files based upon the loader and version provided. So it can only work with what the API provides.

With that said, if you use the `{project}:{versionId}` form it skips the lookup and grabs exactly that file version requested. For example with this command line I can force https://modrinth.com/mod/headbash-commands/version/1.0.0 to be downloaded for NeoForge 1.21.7 even though the mod author set it at 1.21.1:

```
--projects=headbash-commands:6PtEvdAA --loader=neoforge --game-version=1.21.7
```

**Comment by Jannled** [comment]
Wow thank you for the quick response and insight in how the tool works. I thought that I had already pinned down the version like so `essentialsx:2.21.1`, but I didn't expect that I had to use the non human-readable ID, e.g. `essentialsx:V4qyvs91`.

Now everything works^^
